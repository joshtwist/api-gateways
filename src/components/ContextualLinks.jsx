import React from "react";
import DynamicLink from "./common/DynamicLink";
import "./ContextualLinks.scss";

const ContextualLinks = ({ links }) => (
  <div className="contextual-links">
    {links.map((item) => {
      if (item.type === "link") {
        return (
          <div className="contextual-links__link" key={item.name}>
            <DynamicLink url={item.url} name={item.name} />
          </div>
        );
      }
      if (item.type === "subtitle") {
        return (
          <div className="contextual-links__subtitle" key={item.name}>
            {item.name}
          </div>
        );
      }

      return (
        <div className="contextual-links__section" key={Math.random()}>
          {item.name}
        </div>
      );
    })}
  </div>
);

export default ContextualLinks;
