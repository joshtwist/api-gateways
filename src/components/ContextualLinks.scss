@import "./common/_variables.scss";

.contextual-links {
  .contextual-links__section {
    margin-top: 24px;
    margin-bottom: 8px;
    font-size: 16px;
    font-weight: 600;
    color: $grey_80;
  }
  .contextual-links__link {
    margin-bottom: 0;
    padding: 4px 0;
    display: block;

    a {
      color: $grey_50;
      font-size: 14px;
      text-decoration: none;
      border-bottom: 1px solid #ffffff;
      transition: all 0.2s ease-in-out;
    }

    &:hover {
      a {
        color: $blue_60;
        border-bottom: 1px solid $blue_60;
        transition: all 0.2s ease-in-out;
      }
    }
  }

  .contextual-links__subtitle {
    color: $grey-60;
    font-size: 15px;
    font-weight: 600;
    margin-top: 1rem;
    margin-bottom: 0.7rem;
  }

  .contextual-links__alert {
    border: 4px dashed #cce5ff;
    border-radius: 4px 4px 4px 4px;
    padding: 0.75rem 1.25rem;
    color: #0c5460;
  }

  // tablet view
  @media (min-width: 767px) and (max-width: 990px) {
    padding-bottom: 80px;
  }

  // mobile view
  @media (max-width: 767px) {
    padding-bottom: 80px;
  }
}
