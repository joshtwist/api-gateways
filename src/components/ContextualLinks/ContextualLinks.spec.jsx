import React from 'react';
import renderer from 'react-test-renderer';
import ContextualLinks from './ContextualLinks';

describe('ContextualLinks', () => {
  it('renders correctly', () => {
    const testLinks = [
      {
        type: 'section',
        name: 'Prerequisites',
      },
      {
        type: 'link',
        name: 'Download and Install',
        url: 'https://www.postman.com/downloads/',
      },
      {
        type: 'subtitle',
        name: 'Related Blog Posts',

      },
      {
        type: 'link',
        name: 'home',
        url: '/',
      },
    ];

    const tree = renderer
      .create(<ContextualLinks links={testLinks} />)
      .toJSON();
    expect(tree.children[0].props.className).toBe('contextual-links__section');
    expect(tree.children[1].props.className).toBe('contextual-links__link');
    expect(tree).toMatchSnapshot();
  });
});
