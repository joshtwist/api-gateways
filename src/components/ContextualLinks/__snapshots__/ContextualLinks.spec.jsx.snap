// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextualLinks renders correctly 1`] = `
<div
  className="contextual-links"
>
  <div
    className="contextual-links__section"
  >
    Prerequisites
  </div>
  <div
    className="contextual-links__link"
  >
    <a
      className="dynamic-link__external"
      href="https://www.postman.com/downloads/"
    >
      Download and Install
    </a>
  </div>
  <div
    className="contextual-links__subtitle"
  >
    Related Blog Posts
  </div>
  <div
    className="contextual-links__link"
  >
    <a
      className="dynamic-link__internal"
      href="/"
    >
      home
    </a>
  </div>
</div>
`;
